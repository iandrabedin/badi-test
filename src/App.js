import React, { Suspense, lazy } from "react";
import { Navbar, Footer, Loading } from "./components";
import "./app.scss";

const Homepage = lazy(() => import("./pages/homepage"));

function App() {
  return (
    <>
      <Navbar />
      <Suspense fallback={<Loading />}>
        <Homepage />
      </Suspense>
      <Footer />
    </>
  );
}

export default App;
