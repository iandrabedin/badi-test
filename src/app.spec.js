import React from "react";
import { render, cleanup } from "@testing-library/react";
import App from "./app";

afterEach(cleanup);

describe("App", () => {
  it("should render as expected", () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});
