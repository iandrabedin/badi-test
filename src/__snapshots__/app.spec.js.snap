// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render as expected 1`] = `
<DocumentFragment>
  <div
    class="navbar"
  >
    <div
      class="container"
    >
      Recipe Puppy
    </div>
  </div>
  <div
    class="loading"
  >
    Loading...
  </div>
  <footer
    class="footer"
  >
    <div
      class="container"
    >
      Made with
      <a
        href="https://reactjs.org/"
        rel="noopener noreferrer"
        target="_blank"
      >
        ReactJS
      </a>
      and
      <a
        href="http://www.recipepuppy.com/about/api/"
        rel="noopener noreferrer"
        target="_blank"
      >
        Recipe Puppy API
      </a>
    </div>
  </footer>
</DocumentFragment>
`;
